From bc806d8276afc32b22f8e3afe9cfd296356ef45f Mon Sep 17 00:00:00 2001
From: Catalin Popescu <catalin.popescu@intel.com>
Date: Mon, 21 Mar 2016 15:04:40 +0100
Subject: [PATCH 3/7] Dollar Cove PMIC: add GPIO1 regulator.

Declare GPIO1 regulator and add its platform data.

Change-Id: I8a5e73dded831656945566585474b228b990e3ed
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-21105
Signed-off-by: Catalin Popescu <catalin.popescu@intel.com>
Reviewed-on: https://android.intel.com:443/483000
---
 drivers/mfd/intel_soc_pmic_dc.c | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index 25839e9..da77bba 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -26,6 +26,7 @@
 #include <linux/power/dc_xpwr_battery.h>
 #include <linux/power/dc_xpwr_charger.h>
 #include <linux/regulator/intel_dollar_cove_pmic.h>
+#include <linux/regulator/machine.h>
 
 #include <asm/intel_em_config.h>
 #include <linux/extcon/extcon-dc-pwrsrc.h>
@@ -273,6 +274,12 @@ static struct mfd_cell dollar_cove_dev[] = {
 		.num_resources = 0,
 		.resources = NULL,
 	},
+	{
+		.name = "dcovex_regulator",
+		.id = DCOVEX_ID_GPIO1 + 1,
+		.num_resources = 0,
+		.resources = NULL,
+	},
 	{NULL, },
 };
 
@@ -660,6 +667,28 @@ static void dc_set_gpio_pdata(void)
 				sizeof(dollar_cove_gpio_data), 0);
 }
 
+static struct regulator_init_data dcovex_gpio1_data = {
+	.constraints = {
+		.name = "GPIO1",
+		.min_uV = 700000,
+		.max_uV = 3300000,
+		.valid_ops_mask = REGULATOR_CHANGE_VOLTAGE |
+					REGULATOR_CHANGE_STATUS,
+		.valid_modes_mask = REGULATOR_MODE_NORMAL,
+	},
+};
+
+static struct dcovex_regulator_info dollar_cove_reg_gpio1_data = {
+	.init_data = &dcovex_gpio1_data,
+};
+
+static void dc_set_regulator_pdata(void)
+{
+	intel_soc_pmic_set_pdata("dcovex_regulator",
+				(void *)&dollar_cove_reg_gpio1_data,
+				sizeof(dollar_cove_reg_gpio1_data),
+				DCOVEX_ID_GPIO1 + 1);
+}
 
 static int dollar_cove_init(void)
 {
@@ -668,6 +697,7 @@ static int dollar_cove_init(void)
 	dc_xpwr_pwrsrc_pdata();
 	dc_xpwr_fg_pdata();
 	dc_set_gpio_pdata();
+	dc_set_regulator_pdata();
 
 	return 0;
 }
-- 
1.9.1

